vcx/libvcx/src/settings.rs

```diff
@@ -245,6 +250,23 @@ pub fn get_wallet_credentials() -> String {
     credentials.to_string()
 }
 
+pub fn validate_payment_method() -> Result<(), u32> {
+    let config = SETTINGS.read().unwrap();
+    if let Some(method) = config.get(CONFIG_PAYMENT_METHOD) {
+        if !method.to_string().is_empty() {
+            return Ok(());
+        }
+    }
+    return Err(error::MISSING_PAYMENT_METHOD.code_num);
+}
```

Comment:
It seems this function should try to load the payment plugin to test if it is valid, not just if it is present.

Reply:
If the plugin is loaded by the user and can be custom, how should we validate? Three char string?

Reply:
I think we should try to initialize the plugin using the init function.  If that fails then the plugin is invalid.

Reply:
Ticked I was working on states that initialization of the plugin is the responsibility of the user application not libvcx:  https://jira.hyperledger.org/browse/IS-934

Reply:
@glowkey @e-nem-y 

> It seems this function should try to load the payment plugin to test if it is valid, not just if it is present.

Loading can't be responsibility of vcx as it is impossible on the most of platforms. If it is possible we still don't know plugin placement and dll name. 

Initialization can be responsibility of vcx, but in this case we need to provide the name of init method. It looks very unsafe and unnatural. I am not sure that it is possible to call static linked function by name. For loaded plugins it is possible, but on Android and iOS we link statically.

Reply:
One correction. It is possible to call only function by name when we load dll/so from our program, dynamic linking that loads so on program start doesn't allow us it too.

Reply:
If this is the case, that application needs to load plugin itself, then libnullpay (and libsovtoken) will need to create wrappers for the one "init" function.  Correct?

Reply:
In my vision working workflow is the following:

- application links with a plugin or loads it
- application calls init function
- application provides the name of payment method to vcx config

Reply:
No, but for dynamic languages we can provide generic API like:

```
- init_plugin(init_function_name)
- load_plugin(dll path)
```

Reply:
Should that wrapper work be done as part of this PR?

Reply:
I believe it is unnecessary. For example, loading plugin from python or Java is a trivial task (few lines of code). As a result we already can get valuable demo.

Reply:
My main concern is about ConnectMe.